builtin/mainmenu/content/dlg_contentdb.lua

```diff
@@ -0,0 +1,520 @@
+	elseif package.type == "txp" then
+		install_parent = core.get_texturepath()
+	else
+		error("Unknown package type: " .. package.type)
+	end
+
+	if package.queued or package.downloading then
+		return
+	end
+
+	local function on_confirm()
+		local has_hard_deps = contentdb.has_hard_deps(package)
+		if has_hard_deps then
+			local dlg = create_install_dialog(package)
+			dlg:set_parent(this)
+			this:hide()
+			dlg:show()
+		elseif has_hard_deps == nil then
+			local dlg = messagebox("error_checking_deps",
+					fgettext("Error getting dependencies for package"))
+			dlg:set_parent(this)
+			this:hide()
+			dlg:show()
+		else
+			contentdb.queue_download(package, package.path and contentdb.REASON_UPDATE or contentdb.REASON_NEW)
+		end
+	end
+
+	if package.type == "mod" and #pkgmgr.games == 0 then
+		local dlg = messagebox("install_game",
+			fgettext("You need to install a game before you can install a mod"))
+		dlg:set_parent(this)
+		this:hide()
+		dlg:show()
+	elseif not package.path and core.is_dir(install_parent .. DIR_DELIM .. package.name) then
+		local dlg = create_confirm_overwrite(package, on_confirm)
+		dlg:set_parent(this)
+		this:hide()
+		dlg:show()
+	else
+		on_confirm()
+	end
+end
+
+
+-- Resolves the package specification stored in auto_install_spec into an actual package.
+-- May only be called after the package list has been loaded successfully.
+local function resolve_auto_install_spec()
+	assert(contentdb.load_ok)
+
+	if not auto_install_spec then
+		return nil
+	end
+
+	local spec = contentdb.aliases[auto_install_spec] or auto_install_spec
+	local resolved = nil
+
+	for _, pkg in ipairs(contentdb.packages_full_unordered) do
+		if pkg.id == spec then
+			resolved = pkg
+			break
+		end
+	end
+
+	if not resolved then
+		gamedata.errormessage = fgettext("The package $1 was not found.", auto_install_spec)
+		ui.update()
+
+		auto_install_spec = nil
+	end
+
+	return resolved
+end
+
+
+-- Installs the package specified by auto_install_spec.
+-- Only does something if:
+-- a. The package list has been loaded successfully.
+-- b. The ContentDB dialog is currently visible.
+local function do_auto_install()
+	if not contentdb.load_ok then
+		return
+	end
+
+	local pkg = resolve_auto_install_spec()
+	if not pkg then
+		return
+	end
+
+	local contentdb_dlg = ui.find_by_name("contentdb")
+	if not contentdb_dlg or contentdb_dlg.hidden then
+		return
+	end
+
+	install_or_update_package(contentdb_dlg, pkg)
+	auto_install_spec = nil
+end
+
+
+local function sort_and_filter_pkgs()
+	contentdb.update_paths()
+	contentdb.sort_packages()
+	contentdb.filter_packages(search_string, filter_types_type[filter_type])
+
+	local auto_install_pkg = resolve_auto_install_spec()
+	if auto_install_pkg then
+		local idx = table.indexof(contentdb.packages, auto_install_pkg)
+		if idx ~= -1 then
+			table.remove(contentdb.packages, idx)
+			table.insert(contentdb.packages, 1, auto_install_pkg)
+		end
+	end
+end
+
+
+function contentdb.load()
```

Comment:
Since the rest of the `contentdb` table is populated by `contentdb.lua`, it's odd that this function is put into the `contentdb` table.